src/main.tf

```diff
@@ -10,14 +13,14 @@ resource "azurerm_resource_group" "main" {
 }
 
 resource "azurerm_storage_account" "main" {
-  name                              = local.storage_account_name
-  resource_group_name               = azurerm_resource_group.main.name
-  location                          = azurerm_resource_group.main.location
-  account_tier                      = "Standard"
-  account_kind                      = "StorageV2"
-  access_tier                       = "Hot"
-  account_replication_type          = var.redundancy.data_protection ? var.redundancy.replication_type : "LRS"
-  min_tls_version                   = "TLS1_2"
+  name                     = local.storage_account_name
+  resource_group_name      = azurerm_resource_group.main.name
+  location                 = azurerm_resource_group.main.location
+  account_tier             = "Standard"
+  account_kind             = "StorageV2"
+  access_tier              = "Hot"
+  account_replication_type = var.redundancy.data_protection ? var.redundancy.replication_type : local.default_accout_replication_type
```

Comment:
replication_type is required. If `zonal_replication` is not enabled, this still needs to be set. I picked a sane default and added a comment above.